flint/flint-core/src/main/scala/org/opensearch/flint/core/FlintOptions.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.flint.core;
+
+import java.io.Serializable;
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+public class FlintOptions implements Serializable {
+  private Map<String, String> options;
+
+  public static final String INDEX_NAME = "index";
+  public static final String HOST = "host";
+  public static final String PORT = "port";
+  /**
+   * Used by {@link org.opensearch.flint.core.storage.OpenSearchScrollReader}
+   */
+  public static final String SCROLL_SIZE = "scroll_size";
+  public static final int DEFAULT_SCROLL_SIZE = 100;
+
+  public FlintOptions(Map<String, String> options) {
+    this.options = options;
+  }
+
+  public String getIndexName() {
```

Comment:
This option class is in core module but option like `indexName` is actually used by Flint data source layer?

Reply:
both layer using it.
* FlintDataSourceV2 only use it as option parser.
```
  protected def getFlintTable(schema: Option[StructType], option: FlintOptions): FlintTable = {
    FlintTable(option.getIndexName, SparkSession.active, option, schema)
  }
```
* Flint core will use all the option defined.

Reply:
Done. remove depend on FlintOptions from spark-integration module